plugins/modules/aci_l4l7_policy_based_redirect.py

```diff
@@ -0,0 +1,391 @@
+        tenant=dict(type="str", aliases=["tenant_name"]),
+        policy_name=dict(type="str", aliases=["policy"]),
+        description=dict(type="str"),
+        destination_type=dict(type="str", aliases=["dest_type"], choices=["l1", "l2", "l3"]),
+        hash_algorithm=dict(type="str", choices=list(L4L7_HASH_ALGORITHMS_MAPPING)),
+        threshold_enable=dict(type="bool"),
+        max_threshold=dict(type="int"),
+        min_threshold=dict(type="int"),
+        threshold_down_action=dict(type="str", choices=["permit", "deny"]),
+        resilient_hash=dict(type="bool"),
+        pod_aware=dict(type="bool"),
+        anycast_enabled=dict(type="bool"),
+        ip_sla_monitor_policy=dict(type="str", aliases=["monitor_policy", "sla", "sla_policy"]),
+        rewrite_source_mac=dict(type="bool"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["tenant", "policy_name"]],
+            ["state", "present", ["tenant", "policy_name"]],
+        ],
+    )
+    aci = ACIModule(module)
+
+    tenant = module.params.get("tenant")
+    state = module.params.get("state")
+    policy_name = module.params.get("policy_name")
+    description = module.params.get("description")
+    destination_type = module.params.get("destination_type").upper() if module.params.get("destination_type") is not None else None
+    hash_algorithm = L4L7_HASH_ALGORITHMS_MAPPING.get(module.params.get("hash_algorithm"))
+    threshold_enable = aci.boolean(module.params.get("threshold_enable"))
+    max_threshold = module.params.get("max_threshold")
+    min_threshold = module.params.get("min_threshold")
+    threshold_down_action = module.params.get("threshold_down_action")
+    resilient_hash = aci.boolean(module.params.get("resilient_hash"))
+    pod_aware = aci.boolean(module.params.get("pod_aware"))
+    anycast_enabled = aci.boolean(module.params.get("anycast_enabled"))
+    ip_sla_monitor_policy = module.params.get("ip_sla_monitor_policy")
+    rewrite_source_mac = aci.boolean(module.params.get("rewrite_source_mac"))
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="vnsSvcRedirectPol",
+            aci_rn="svcCont/svcRedirectPol-{0}".format(policy_name),
+            module_object=policy_name,
+            target_filter={"name": policy_name},
+        ),
+        child_classes=["vnsRsIPSLAMonitoringPol"],
+    )
+
+    aci.get_existing()
+
+    if state == "present":
+        child_configs = []
+        if ip_sla_monitor_policy:
+            monitor_tdn = "uni/tn-{0}/ipslaMonitoringPol-{1}".format(tenant, ip_sla_monitor_policy)
```

Comment:
Is it possible to add an IPSLA monitoring policy from a different tenant? I presume not, but thought of double checking.

Reply:
Nope!